source/_posts/vue-design-patterns.md

```diff
@@ -0,0 +1,988 @@
+        };
+    
+        return {
+          data: computed(() => todoStore.state.todo.list),
+          addTodo,
+        };
+      },
+    });
+    </script>
+```
+
+The Pinia version looks like this.
+
+```ts
+    // File: PiniaTodoList.vue
+
+    <template>
+      <TodoList :data="todoList" />
+      <button @click="addTodo">Add Todo</button>
+    </template>
+    
+    <script lang="ts">
+    import { computed, defineComponent, onMounted } from "vue";
+    import { useTodoStore } from "@/store/pinia/todo";
+    import TodoList from "@/components/VList.vue";
+    
+    export default defineComponent({
+      components: { TodoList },
+      setup() {
+        const store = useTodoStore();
+        const { fetchTodos, addTodo: newTodo } = store;
+        const todoList = computed(() => store.getList);
+    
+        const addTodo = () => {
+          newTodo({
+            title:
+              "This is a random todo number " + Math.random().toFixed(2).toString(),
+            completed: Math.random() > 0.5 ? false : true,
+          });
+        };
+        onMounted(fetchTodos);
+    
+        return { todoList, addTodo };
+      },
+    });
+    </script>
+```
+
+Both work well, but the superficial differences are few. But Pinia hides something really interesting. We are using Pinia as a Composable. Remember, the principal mission of this pattern is to manage and expose the state in a good way. Naturally, Pinia does that and fits in. but the Pinia approach has another advantage, Type friendly. We don't need to point specific stores, actions, mutations, getters, or others via strings (a non-type friend); it happens in Vuex. In the past, that detail always caused me problems with the code. Instead, in Pinia, each state, action, or getter will be accessible directly from the source as a `useStoreSomething()`. Pinia is the new by-default state management library for Vue. We recommend using it belong the composition API.
+
+> Code:  [vue design pattern repository](https://github.com/cr0wg4n/vue-design-patterns), execute, and go to **/store-management-pattern** route 😎.
+
+## Summary
+Writing this article was a big challenge for me. I hope these examples and guideline helps you to make the best architectural decisions. Getting motivated and up to date is hard. If you found it useful, don't doubt to share this article with your #VueFriends.
+
+I don't want to say goodbye without first recommending the Lachlan Miller's [book](https://lachlan-miller.me/design-patterns-for-vuejs) called "Design Pattern for Vue.js. A Test-driven Approach to maintainable Applications" which inspired me to write this article and the magnificent [conference](https://youtu.be/RF1bbhRw9sg) of Jacob Schatz (you will find more advanced patterns with Vue).
+
+As we mentioned earlier, all the code is available [here](https://github.com/cr0wg4n/vue-design-patterns). Clone, execute, and enjoy it. Says goodbye, your #VuenFriend [@cr0wg4n](https://twitter.com/cr0wg4n).
```

Comment:
`#VuenFriend` ? 🥲

Reply:
VueFriend was the goal. For Latin American Vue people is ok to use "Vuen" as "Buen" (in Spanish, Good, so Good Friend). But I think `#VueFriend` is better.